x/restake/keeper/keeper_test.go

```diff
@@ -0,0 +1,259 @@
+	ValidAddress2 = sdk.AccAddress("1000000002")
+	ValidAddress3 = sdk.AccAddress("1000000003")
+
+	ActiveVaultKey   = "active_vault_key"
+	InactiveVaultKey = "inactive_vault_key"
+	InvalidVaultKey  = "invalid_key"
+
+	LiquidStakerAddress = sdk.AccAddress("12345678901234567890123456789012")
+
+	ValAddress = sdk.ValAddress("4000000001")
+)
+
+type KeeperTestSuite struct {
+	suite.Suite
+
+	ctx           sdk.Context
+	storeKey      storetypes.StoreKey
+	restakeKeeper keeper.Keeper
+	accountKeeper *restaketestutil.MockAccountKeeper
+	bankKeeper    *restaketestutil.MockBankKeeper
+	stakingKeeper *restaketestutil.MockStakingKeeper
+
+	stakingHooks stakingtypes.StakingHooks
+
+	queryClient types.QueryClient
+	msgServer   types.MsgServer
+
+	validVaults []types.Vault
+	validLocks  []types.Lock
+	validParams types.Params
+	validStakes []types.Stake
+}
+
+func TestKeeperTestSuite(t *testing.T) {
+	suite.Run(t, new(KeeperTestSuite))
+}
+
+func (suite *KeeperTestSuite) SetupTest() {
+	suite.validParams = types.NewParams([]string{"uband"})
+	suite.validStakes = []types.Stake{
+		{
+			StakerAddress: ValidAddress1.String(),
+			Coins:         sdk.NewCoins(sdk.NewCoin("uband", sdkmath.NewInt(50))),
+		},
+		{
+			StakerAddress: ValidAddress3.String(),
+			Coins:         sdk.NewCoins(sdk.NewCoin("uband", sdkmath.NewInt(10))),
+		},
+	}
+	suite.validVaults = []types.Vault{
+		{
+			Key:      ActiveVaultKey,
+			IsActive: true,
+		},
+		{
+			Key:      InactiveVaultKey,
+			IsActive: false,
+		},
+	}
+
+	suite.validLocks = []types.Lock{
+		{
+			StakerAddress: ValidAddress1.String(),
+			Key:           ActiveVaultKey,
+			Power:         sdkmath.NewInt(100),
+		},
+		{
+			StakerAddress: ValidAddress1.String(),
+			Key:           InactiveVaultKey,
+			Power:         sdkmath.NewInt(50),
+		},
+		{
+			StakerAddress: ValidAddress2.String(),
+			Key:           ActiveVaultKey,
+			Power:         sdkmath.NewInt(10),
+		},
+	}
+
+	key := storetypes.NewKVStoreKey(types.StoreKey)
+	suite.storeKey = key
+	testCtx := testutil.DefaultContextWithDB(suite.T(), key, storetypes.NewTransientStoreKey("transient_test"))
+	suite.ctx = testCtx.Ctx.WithBlockHeader(tmproto.Header{Time: tmtime.Now()})
+	encCfg := moduletestutil.MakeTestEncodingConfig()
+	moduleAccount := authtypes.NewEmptyModuleAccount(types.ModuleName)
+
+	// gomock initializations
+	ctrl := gomock.NewController(suite.T())
+	accountKeeper := restaketestutil.NewMockAccountKeeper(ctrl)
+	accountKeeper.EXPECT().
+		GetModuleAddress(types.ModuleName).
+		Return(moduleAccount.GetAddress()).
+		AnyTimes()
+	accountKeeper.EXPECT().
+		GetModuleAccount(gomock.Any(), types.ModuleName).
+		Return(moduleAccount).
+		AnyTimes()
+	accountKeeper.EXPECT().
+		SetModuleAccount(gomock.Any(), moduleAccount).
+		Return().
+		AnyTimes()
+	suite.accountKeeper = accountKeeper
+
+	bankKeeper := restaketestutil.NewMockBankKeeper(ctrl)
+	bankKeeper.EXPECT().
+		GetAllBalances(gomock.Any(), moduleAccount.GetAddress()).
+		Return(nil).
+		Times(1)
```

Comment:
Why just one time?

Reply:
this expectation is for initgenesis on setup only. after that, the balance might be changed.